test/parallel/test-stream-pipe-cleanup-pause.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+var common = require('../common');
+var assert = require('assert');
+var stream = require('stream');
+
+var reader = new stream.Readable();
+var writer1 = new stream.Writable();
+var writer2 = new stream.Writable();
+var writer1Writes = 0;
+var writer2Writes = 0;
+
+reader._read = function(n) {};
+
+writer1._write = function(chunk, encoding, cb) {
+  ++writer1Writes;
+  this.emit('chunk-received');
+  cb();
+};
+writer1.once('chunk-received', function() {
+  reader.unpipe(writer1);
+  reader.pipe(writer2);
+  reader.push(new Buffer(560000));
+  setImmediate(function() {
+    reader.push(new Buffer(560000));
+    setImmediate(function() {
+      reader.push(new Buffer(560000));
+    });
+  });
+});
+
+writer2._write = function(chunk, encoding, cb) {
+  ++writer2Writes;
+  cb();
+};
+
+reader.pipe(writer1);
+reader.push(new Buffer(560000));
+
+process.on('exit', function() {
```

Comment:
`common.mustCall` will only help so much since I also want to verify the number of times the function was called.